sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs

```diff
@@ -183,6 +184,32 @@ private static Diagnostic CreateDiagnosticFromJToken(JToken token)
 
             return Diagnostic.Create(rule, loc);
         }
+        private static Diagnostic CreateDiagnosticFromFailureString(string filename, string description)
+        {
+            var rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, description, RspecStrings.ResourceManager);
+
+            var begin = new LinePosition(1,1);
+            var end = new LinePosition(1,2);
+            var loc = Location.Create(filename, TextSpan.FromBounds(0, 0), new LinePositionSpan(begin, end));
+
+            return Diagnostic.Create(rule, loc);
+        }
+        private void RaiseIssueFromFailedCbdeRun(CompilationAnalysisContext context)
+        {
+            string failureString = "CBDE Failure Report :\n  C# souces files involved are:\n";
+            foreach (var fileName in csSourceFileNames)
```

Comment:
Use `string.join`

Reply:
I am just using the StringBuilder as suggested below as re-configuring the code to use string.Join would be far less readable for a negligible perf boost.